### PR TITLE
Qt\Settings-WiiPane: Turn PAL60 into 50/60Hz-mode radio buttons

### PR DESCRIPTION
**Pull request by ghost**
Redo of #2618 in Qt

pal60_refreshrate_radiobuttons_v1

### CODE REVIEW

File: Source/Core/DolphinQt/Settings/WiiPane.cpp [hunk 1]
```diff
@@ -99,7 +103,19 @@ void WiiPane::CreateMisc()
...
-  m_pal60_mode_checkbox = new QCheckBox(tr("Use PAL60 Mode (EuRGB60)"));
+
+  auto* refresh_rate_pal_box = new QGroupBox(tr("Refresh Rate (PAL)"));
+  auto* refresh_rate_pal_layout = new QHBoxLayout;
+
+  refresh_rate_pal_box->setLayout(refresh_rate_pal_layout);
+  m_refresh_rate_pal_mode_50hz = new QRadioButton(tr("50 Hz"));
+  m_refresh_rate_pal_mode_60hz = new QRadioButton(tr("60 Hz"));
+
+  // refresh_rate_pal_layout->addStretch(1);
```
Comment (BhaaLseN) [inline Source/Core/DolphinQt/Settings/WiiPane.cpp]: Please don't leave commented out code lying around like this. Either remove it completely, or use it.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp [hunk 2]
```diff
@@ -99,7 +103,19 @@ void WiiPane::CreateMisc()
...
+  refresh_rate_pal_layout->addWidget(m_refresh_rate_pal_mode_50hz);
+  refresh_rate_pal_layout->addWidget(m_refresh_rate_pal_mode_60hz);
+  // refresh_rate_pal_layout->addStretch(1);
```
Comment (BhaaLseN) [inline Source/Core/DolphinQt/Settings/WiiPane.cpp]: Same down here.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp [hunk 3]
```diff
@@ -189,22 +195,69 @@ void WiiPane::CreateWiiRemoteSettings()
   wii_remote_settings_group_layout->addWidget(m_wiimote_motor, 3, 0, 1, -1);
 }
 
-void WiiPane::OnEmulationStateChanged(bool running)
+void WiiPane::CreateWhitelistedUSBPassthroughDevices()
```
Comment (BhaaLseN) [inline Source/Core/DolphinQt/Settings/WiiPane.cpp]: Why did you move this code around? It has no relevance to your PR.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp [hunk 4]
```diff
@@ -189,22 +195,69 @@ void WiiPane::CreateWiiRemoteSettings()
...
+  SConfig::GetInstance().m_usb_passthrough_devices.erase({vid_u16, pid_u16});
+  PopulateUSBPassthroughListWidget();
 }
 
 void WiiPane::LoadConfig()
 {
   m_screensaver_checkbox->setChecked(Config::Get(Config::SYSCONF_SCREENSAVER));
-  m_pal60_mode_checkbox->setChecked(Config::Get(Config::SYSCONF_PAL60));
+
+  if (Config::Get(Config::SYSCONF_PAL60) == true)
```
Comment (BhaaLseN) [inline Source/Core/DolphinQt/Settings/WiiPane.cpp]:
```suggestion
  if (Config::Get(Config::SYSCONF_PAL60))
```

File: Source/Core/DolphinQt/Settings/WiiPane.cpp (hunk 3 above)
Comment (Zexaron) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: It's just confusing a bit when similar panes have different methods of achieving the same exact thing, such as method names and sorting. Yes this would fit better in a code cleanup PR.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp [hunk 5]
```diff
@@ -120,14 +135,21 @@ void WiiPane::CreateMisc()
   m_system_language_choice->addItem(tr("Traditional Chinese"));
   m_system_language_choice->addItem(tr("Korean"));
 
-  m_pal60_mode_checkbox->setToolTip(tr("Sets the Wii display mode to 60Hz (480i) instead of 50Hz "
-                                       "(576i) for PAL games.\nMay not work for all games."));
+  m_refresh_rate_pal_mode_50hz->setToolTip(
+      tr("Sets the Wii display mode to 50Hz (576i) instead of 60Hz "
```
Comment (MayImilae) [inline Source/Core/DolphinQt/Settings/WiiPane.cpp]: It is useful to repeat that this is for PAL games only here in the tooltip, in case someone glances over the PAL title. Also some variations can help with making it a bit more readable. Something like "Sets the Wii display mode to the standard 50Hz (576i) mode for PAL games."

Also I wasn't able to find a specific list, but the number of PAL games that do not support 50hz can be counted on one hand. If you'd like to mention it, please say something like "Almost all PAL titles use this mode."

All together, I recommend: 

>Sets the Wii display mode to the standard 50Hz (576i) mode for PAL games. Almost all PAL titles use this mode.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp [hunk 6]
```diff
@@ -120,14 +135,21 @@ void WiiPane::CreateMisc()
...
+         "(480i).\nSome games do not support 50Hz mode."));
+  m_refresh_rate_pal_mode_60hz->setToolTip(
+      tr("Sets the Wii display mode to 60Hz (480i) instead of 50Hz "
```
Comment (MayImilae) [inline Source/Core/DolphinQt/Settings/WiiPane.cpp]: This one is good, though I'd recommend adding the word standard to this for clarity, as well as mentioning PAL once again. Like this:

>Sets the Wii display mode to 60hz (480i) instead of the standard 50hz (576i) for PAL games. Only works with PAL titles that support it.

**Comment by Helios747** [comment]
What's wrong with the checkbox again?

File: Source/Core/DolphinQt/Settings/WiiPane.cpp (hunk 6 above)
Comment (Zexaron) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: Hah, that was literally the first iteration I had until I chanded, reverting.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp (hunk 5 above)
Comment (Zexaron) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: I believe Wii autodetects EDTV/HDTV as I was playing SSBB on Wii just hours ago (after many years, it's still alive!) and when I tried running a demo classic inside SSBB it asked me it had to swith the display mode to 50hz, so I would replace the "use" with "support" as "use" might imply it's default irrespective of connection type or TV.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp (hunk 6 above)
Comment (Zexaron) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: Weird, why more "PAL". I removed the "for PAL games" on purpose because it was too much. rrrright?

**Comment by Zexaron** [comment]
@Helios747 Which one, the old PAL60 or the SD Card alignment? 

IMO both SD Card and USB Keyboard should go up a bit so they're in line with the radio box. But I will give a few options to preview, but more later, need to figure out more about how alignments work.

**Comment by MayImilae** [comment]
@Helios747 Zexaron took issue with the name of the option (PAL60) being a name the Wii does not use.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp (hunk 5 above)
Comment (JosJuice) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: Yes, it should say support rather than use. Though I can't recall it switching to 60 Hz automatically like you say.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp (hunk 6 above)
Comment (JosJuice) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: Should we mention that titles that don't support it will fall back to 50 Hz? Just so nobody thinks they should avoid PAL60 for compatibility reasons.

**Comment by Zexaron** [comment]
Oh yeah, I should have reviewd the background behind this change: @Helios747 I'm not actually against PAL60/EuRGB60 monikers, however PAL Wii does not mention them anywhere on screen or in the booklets as far as I can tell, I have opened the cases up, played some games on it to check (it's literally "50hz (576i) mode"), so that's one reason but a minor one, the major reason is that the setting didn't imply that it is "PAL-only" and making it convey that context would have made it much longer and also sound weird with so many repeating words, that PR has since been closed but others pointed out an old PR for Wx, in which it was easy to add "(PAL only)" at the back of the title, and solving the whole thing. 

Adding "PAL60" and/or "EuRGB60" would now be kinda redundant and probably too saturated.

**Comment by Zexaron** [comment]
So here's the 2 possibilities of alignment.
pal60_refreshrate_radiobuttons_v2a
pal60_refreshrate_radiobuttons_v2b

**Comment by JosJuice** [comment]
Both of those look pretty weird. It would probably be the best to not have a checkbox next to the radio buttons.

**Comment by Dockolm** [comment]
What about regrouping all three checkboxes together on the left, and put the radio buttons on the right ?

**Comment by Zexaron** [comment]
@Dockolm  Yeah that's one way of doing it, It's just that the radio buttons box affects the row width for other columns as well so number 2 goes below radio buttons and number 1 is centered in the middle.

Now I put them all on one slot and used additional alignment. Option B is the current code.

pal60_refreshrate_radiobuttons_v3a
pal60_refreshrate_radiobuttons_v3b

File: Source/Core/DolphinQt/Settings/WiiPane.cpp (hunk 5 above)
Comment (Zexaron) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: Allright, I assumed it's 60Hz because when I try to play trial versions of Masterpieces in SSBB it says it needs to switch to 50hz, , ofcourse that's not proof.

But I forgot to mention, I am connecting via COMPONENT cable (YPbPr) and I have checked right now to make sure and yes the HDTV (Samsung UE40D6750) is showing 720x480 60hz on the source label as the Wii warning screen is showing when it boots.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp (hunk 6 above)
Comment (Zexaron) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: Sure, but I haven't seen the fallback code myself, didn't came across it. Are you sure it fallbacks?

Comment (AdmiralCurtiss) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: The way it works on console is that you set a video mode in the System Menu, then the System Menu boots the game with whatever the current VI registers are (so the video mode you have configured), and then the game is free to react on either the VI registers or try to parse the configuration itself. So all games have to in some way handle all possible options.

Official releases that have been licensed by Nintendo are required to support 50 Hz in some form, even if it's just displaying a "This game can't run in 50 Hz, please exit the game and change the output to 60 Hz" message -- some late Wii releases do this. On the other hand, no game is required to support 60 Hz at all, it's perfectly legal for a game to ignore the 60 Hz setting and switch to 50 Hz on boot regardless of the VI or system settings.

So yes, I can guarantee it falls back to 50 Hz if 60 Hz is not supported.


As a sidenote, if your system can handle the 10 Hz difference, running in 60 will give you an equivalent or better experience in almost every game; lots of games were still poorly converted to 50 Hz even as late as the Wii's lifecycle. So I support phrasing this in a way that encourages people to set the setting to 60 Hz.

File: Source/Core/DolphinQt/Settings/WiiPane.cpp (hunk 5 above)
Comment (JosJuice) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: You've probably changed the setting to 60 Hz at some point.

**Comment by MayImilae** [comment]
Option A is definitely better, imo, since it has more separation. But it looks much better!

**Comment by Zexaron** [comment]
Figured more out now - I can make vertical separation between them wider if needed too.

pal60_refreshrate_radiobuttons_v41

File: Source/Core/DolphinQt/Settings/WiiPane.cpp (hunk 5 above)
Comment (Zexaron) [reply Source/Core/DolphinQt/Settings/WiiPane.cpp]: Oops. Indeed. I forgot quite a bit I guess. Wii has 3 options and I have it set to EDTV/HDTV 480p which is 60hz. But there is also a 60hz 480i mode.

I think I heard dolphin doesn't emulate interlaced right. so those, but if the PAL60 (E60 in code) is the SYSCONF Wii native settings, which of the two is it?

**Comment by Zexaron** [comment]
I experimented some more. It didn't pan out good in my attempts, putting the 3 checkboxes into one row, as well as the radio buttons (while inside the box). 

But I made a slight adjustment to the A option like so:

pal60_refreshrate_radiobuttons_v42

**Comment by 8times9** [comment]
Zexaron deleted their GitHub account, odd.
